[Cheong Li Yuan] Duke Increments

cheongliyuan opened the pull request:
Looking forward to your reviews!

j-lum committed:
Add support for Gradle workflow

j-lum committed:
Add sample checkstyle configuration

Add toolVersion block in to Gradle code sample to prevent errors.

j-lum committed:
Change file mode on `gradle` to be executable

j-lum committed:
Merge pull request #12 from j-lum/gradle+x

Change file mode on `gradle` to be executable (#9)

j-lum committed:
Add configuration for console applications

Gradle defaults to an empty stdin which results in runtime exceptions
when attempting to read from `System.in`. Let's add some sensible
defaults for students who may still need to work with the standard
input stream.

j-lum committed:
Merge pull request #13 from j-lum/javaexec

Add configuration for console applications

cheongliyuan committed:
Merge branch 'gradle'

cheongliyuan committed:
Level 1

cheongliyuan committed:
level 2

cheongliyuan committed:
feeling abit uncertain without implementing try catch..

cheongliyuan committed:
fixed some mistakes

cheongliyuan committed:
Remove old parsers (inputIsBye() etc) and combine into addTask()
Add 3 subclasses for each type of task E/D/T
Swap ifelse in main for switchcase, as there is no need for priority
Add enum Tasktype for parsing purposes
Swap for loop for collection Tasktype userTask
Add numOfTasks to help keep track

cheongliyuan committed:
Missed out some final cleaning
Cleanup test println

cheongliyuan committed:
bug fixes

cheongliyuan committed:
Revert "bug fixes"

This reverts commit 2b63b88b828d40550ef7058dd6ecaf77e6292161.

cheongliyuan committed:
bug fixes

cheongliyuan committed:
Add minimal exceptions + 'done' case

cheongliyuan committed:
Add deleteTask
Add exception for deleteTask
Fix some printf errors

cheongliyuan committed:
Move methods around for readability
Not confident on public/private access...

cheongliyuan committed:
Shift methods around to improve readability
Prepare for branching

cheongliyuan committed:
Change access modifier for some methods from public to package private
Add ability to save list to txt file.
Add updateSaveFile for various user inputs(addTask/Delete/Done)
Add generateListForFile and writeStringToFile as supporting methods
Change getStatusIcons to '1' and '0'. Unable to see symbols in both intellji and txt file.

shihaoyap commented:
Hi Li Yuan! Thank you for your comments on my program! :) I got the chance to test your program and have found some places that you could look into!

1) Suggest using another symbol instead of the unicode since we cant see symbols in intellji. -  Suggest using X , O instead, it will be easier :)

2) ExceptionHandling for special cases could be included! 
     a) When you do "done 2", to a list with only 1 task in list, no output / exception is given.  
     b) When you try to delete a invalid index, and the index is not filled in the list - error is thrown
  
3) Suggest to split the class files to different files for better organisation (might be my own preference)

4) Exception class - UnknownInputException and BadInputException class could be under the same class

Overall, the program was well written! Good practice of OOP fundamental principles. Code readability was good and code style was also done properly. Maybe you could include some javaDocs to the more complicated methods and class files as well for better documentation. 

Good job! Good luck for the increments ahead and i look forward to seeing more! 👍

cheongliyuan committed:
Add DateTimeFormatter to parse date and time
add DateTimeParseDukeException to catch wrong date input formats

cheongliyuan committed:
Merge branch 'branch-Level-7'

cheongliyuan committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

CarbonGrid commented:
Good afternoon @cheongliyuan ,

Good effort for having completed Level-8 so far.

However, there is quite alot room for improvements in your code. 👀 

You could consider:
- Making individual top-class classes public.
- Enable checkstyle gradle plugin and adopt a common coding standard.
- Avoid using "EXIT_COMMAND" to name non final static variables.
- DateTimeParseDukeException doesn't display boundary lines as per most prints do.
- Duke doesn't load / auto-load "duke.txt" as per Level-7 requirements.
- Reduce code base by factoring common methods or variables into parent methods 
- Try to reduce hard coding & code base by defining methods that are used commonly
- Restrictive parsing " list" throws an error
- Try to adopt OOP and segment them into individual java class files and restrict access modifiers to only what is needed.
- "done -1" or "done 2" when list is size 1, silently fail.
- "delete -1" or "delete 3" unhandled Exception when size is only 2.
- Try to use JUnit tests.
- Remove unused variables.

And just to state I noticed, really great job for citing other's work. 👍 
It shows the honest efforts you put in but I would suggest you to approach your teammates or fellow classmate for help. Learning on your own means you need to know what you don't know and learn those yourself hence moving at speed x1, seeking knowledge with abit of help means others showing what you don't know and you can get straight to learning what you don't know hence moving at speed x2 or more. 

I'm not sure how busy you are but good luck in your time management. The amount of time you spend on this corresponds to the amount you have to learn.

With experience, you could achieve all requirements with minimal coding.

Anyway good job for the honest & sincere effort and I hope you can keep up.

CarbonGrid reviewed (commented):
Good afternoon @cheongliyuan ,

Good effort for having completed Level-8 so far.

However, there is quite alot room for improvements in your code. 👀

You could consider:

- Making individual top-class classes public.
- Enable checkstyle gradle plugin and adopt a common coding standard.
- Avoid using "EXIT_COMMAND" to name non final static variables.
- DateTimeParseDukeException doesn't display boundary lines as per most prints do.
- Duke doesn't load / auto-load "duke.txt" as per Level-7 requirements.
- Reduce code base by factoring common methods or variables into parent methods
- Try to reduce hard coding & code base by defining methods that are used commonly
- Restrictive parsing " list" throws an error
- Try to adopt OOP and segment them into individual java class files and restrict access modifiers to only what is needed.
- "done -1" or "done 2" when list is size 1, silently fail.
- "delete -1" or "delete 3" unhandled Exception when size is only 2.
- Try to use JUnit tests.
- Remove unused variables.

And just to state I noticed, really great job for citing other's work. 👍
It shows the honest efforts you put in but I would suggest you to approach your teammates or fellow classmate for help. Learning on your own means you need to know what you don't know and learn those yourself hence moving at speed x1, seeking knowledge with abit of help means others showing what you don't know and you can get straight to learning what you don't know hence moving at speed x2 or more.

I'm not sure how busy you are but good luck in your time management. The amount of time you spend on this corresponds to the amount you have to learn.

With experience, you could achieve all requirements with minimal coding.

Anyway good job for the honest & sincere effort and I hope you can keep up.

cheongliyuan committed:
Add and split up command files
Add and split up exception files
Change Task to abstract class
Split up D/E/T to different files

cheongliyuan committed:
Add test files

cheongliyuan committed:
updated gradle

cheongliyuan committed:
Add JavaDoc for files

cheongliyuan committed:
Fix spacing
Remove unneeded imports

cheongliyuan committed:
Add FindCommand class
Add findKeyWordInList method in Parser

cheongliyuan committed:
Merge branch 'branch-A-JavaDoc' into Level-9

# Conflicts:
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/command/ReadCommand.java

cheongliyuan committed:
Merge branch 'A-CodingStandard' into Level-9

# Conflicts:
#	src/main/java/duke/command/ReadCommand.java

cheongliyuan committed:
Touchups before A-gradle

cheongliyuan committed:
Mistake: Merged A-codingStandard and branch-A-JavaDoc with Level-9 instead of master. Reverse commit here to Merge everything with master.

cheongliyuan committed:
Merge branch 'Level-9'

cheongliyuan committed:
Fix formatting issues.
Checkstyle so goodd

cheongliyuan committed:
Add shadowJar for quick compiling. This was done previously.

cheongliyuan committed:
checkpoint: GUI tutorial

cheongliyuan committed:
Add GUI interface. (Minimal)
Todo: Fix Dialogbox allow expansion to fit larger list.
Change AddCommand ui.Show formatting to fit GUI. (Combine into 1 ui.show function)
Add MainUi, child class of Ui
Add DialogBox.fxml ,MainWindow.fxml
Add getResponse in Duke to pass messages between Ui and Duke.
Fix exception string formatting to fit GUI

cheongliyuan committed:
Merge branch 'A-CheckStyle'

cheongliyuan committed:
Merge branch 'A-Gradle'

cheongliyuan committed:
Merge branch 'Level-10'

# Conflicts:
#	src/main/java/duke/exception/BadInputException.java
#	src/main/java/duke/exception/EmptyDateTimeDukeException.java
#	src/main/java/duke/exception/UnknownInputException.java

cheongliyuan committed:
Add assertions
Retry PR

cheongliyuan committed:
Add assertions
Retry PR

cheongliyuan committed:
Merge branch 'master' of https://github.com/cheongliyuan/duke into A-Assertions

cheongliyuan committed:
Re-retry PR

cheongliyuan committed:
Re-retry PR (#3)

master s&m

cheongliyuan committed:
Merge branch 'master' of https://github.com/cheongliyuan/duke

* 'master' of https://github.com/cheongliyuan/duke:
  Re-retry PR (#3)

cheongliyuan committed:
Remove unused Exception
Fix new Checkstyle errors
Not sure what to refactor

cheongliyuan committed:
Re-PR because sourcetree fails git PR when 'tags are clashing'

cheongliyuan committed:
Merge branch 'A-CodeQuality'

cheongliyuan committed:
Add ability to Tag tasks
Change generateListForFile to accomodate saving of Tag
Change parser to read Tags (/t tagName)
Add getTagName to Task for file saving purposes
Change GUI: TextField and SendButton to anchor to window at side
Update task.txt with proper example tasks

cheongliyuan committed:
MIstake: commited instead of git push. Retrying PR

cheongliyuan committed:
Add some GUI improvements

cheongliyuan committed:
Update README.md
Add Ui.PNG

cheongliyuan committed:
Set theme jekyll-theme-midnight

cheongliyuan committed:
Set theme jekyll-theme-hacker

cheongliyuan committed:
Set theme jekyll-theme-hacker

cheongliyuan committed:
fix minor format issues

cheongliyuan committed:
Rename Ui file

cheongliyuan committed:
Merge branch 'master' of https://github.com/cheongliyuan/duke

what is there?

cheongliyuan committed:
fix ui.png

cheongliyuan committed:
1

cheongliyuan committed:
2

cheongliyuan committed:
Final release
Credits to CarbonGrid! (See Launcher.java)
